tests/test_bot.py

```diff
@@ -1433,14 +1433,10 @@ def test_camel_case_aliases(self, bot_class, bot_method_name, bot_method):
         assert camel_case_function is bot_method, f"{camel_case_name} is not {bot_method}"
 
     @bot_methods()
-    def test_coroutine_functions(self, bot_class, bot_method_name, bot_method):
+    def test_coroutine_functions(self, bot_class, bot_method_name, bot_method, monkeypatch):
         """Check that all bot methods are defined as async def  ..."""
-        # unfortunately `inspect.iscoroutinefunction` doesn't do the trick directly,
-        # as the @_log decorator interferes
-        source = "".join(inspect.getsourcelines(bot_method)[0])
-        assert (
-            f"async def {bot_method_name}" in source
-        ), f"{bot_method_name} should be a coroutine function"
+        meth = getattr(bot_method, "__wrapped__", bot_method)
```

Comment:
please leave a CSI comment on why we need the `__wrapped__`.

Reply:
added in https://github.com/python-telegram-bot/python-telegram-bot/pull/3426/commits/b2e292ea6bd24e57d7cdd92dc169843868871411